### ISSUE TITLE
Activity Logs - Developer Portal

### ISSUE THREAD

**Issue by campifoo** [Azure/api-management-developer-portal#1208]
Activity Logs do not capture Subscription Activity When Happens in Developer Portal

When I access the Subscription blade in Azure portal and create a subscription and regenerate keys, the activity is logged in the Activity Logs.  When I do the same activity using the developer portal, I do not see the activity in the logs.   Is there a way to ensure the Activity is captured when the subscription is created and keys are regenerated when a developer uses the dev portal?

**Comment by mikebudzynski** [comment]
Administrative actions are automatically included in the Activity Logs, because the API calls are routed through Azure Resource Manager, while developer portal actions are routed directly to API Management. At this moment, we're unsure about the scope of this improvement and can't provide an ETA.

**Comment by AnRei123** [comment]
@mikebudzynski: Do you plan to log user activities with the users ID, for example, when the user logs in to the DevPortal? We like to keep track on when a user has logged in the last time to the DevPortal. If a user that has been registered with a valid AAD account hasn´t logged in to the DevPortal for more than one year, we like to let him confirm if the account is still required or not.  A prerequisite is that we are able to check an activity log for a DevPortal and check for events for a dedicated users id. Should I create a new ticket for this request or would this requirement already covered by your upcoming monitoring activities?

**Comment by mikebudzynski** [comment]
> Do you plan to log user activities with the users ID, for example, when the user logs in to the DevPortal?

You integrate Application Insights or Google Analytics into the portal to track the log-in actions.

> If a user that has been registered with a valid AAD account hasn´t logged in to the DevPortal for more than one year

I created a new feature request #1283 for providing developer ecosystem telemetry.

**Comment by Sachin4dotnet** [comment]
Can you give some custom URL to be triggered when someone subscribe the product? That URL must be configurable from portal.

**Comment by mikebudzynski** [comment]
Yes, you can use the delegation functionality: https://docs.microsoft.com/en-us/azure/api-management/api-management-howto-setup-delegation

**Comment by DanMansfield520** [comment]
Has a date been established to address the subscription log activity? Haven't seen any updates since last year.

**Comment by DanMansfield520** [comment]
> Administrative actions are automatically included in the Activity Logs, because the API calls are routed through Azure Resource Manager, while developer portal actions are routed directly to API Management. At this moment, we're unsure about the scope of this improvement and can't provide an ETA.
> 
> In the coming months, we plan to provide support for events in API Management. You will be able to subscribe to an event feed and monitor it for actions such as rotation of subscription keys.

Has a date been established to address the subscription log activity? Haven't seen any updates since last year.

**Comment by DanMansfield520** [comment]
@mikebudzynski Has a date been established to address the subscription log activity? Haven't seen any updates since last year.
